src/Events/Admin/Onboarding/API.php

```diff
@@ -0,0 +1,220 @@
+<?php
+/**
+ * The REST API handler for the Onboarding Wizard.
+ * Cleverly named...API.
+ *
+ * @since 7.0.0
+ *
+ * @package TEC\Events\Admin\Onboarding
+ */
+
+namespace TEC\Events\Admin\Onboarding;
+
+use WP_REST_Request as Request;
+use WP_REST_Server as Server;
+use WP_Error;
+use WP_REST_Response;
+use TEC\Events\Admin\Onboarding\Data;
+
+/**
+ * Class API
+ *
+ * @since 7.0.0
+ *
+ * @package TEC\Events\Admin\Onboarding
+ */
+class API {
+
+	/**
+	 * The action for this nonce.
+	 *
+	 * @since 7.0.0
+	 *
+	 * @var string
+	 */
+	public const NONCE_ACTION = '_tec_wizard';
+
+	/**
+	 * Rest Endpoint namespace
+	 *
+	 * @since 7.0.0
+	 *
+	 * @var  string
+	 */
+	protected const ROOT_NAMESPACE = 'tec/v2/onboarding';
+
+	/**
+	 * Register the endpoint.
+	 *
+	 * @since 7.0.0
+	 *
+	 * @return bool If we registered the endpoint.
+	 */
+	public function register(): bool {
+		return register_rest_route(
+			self::ROOT_NAMESPACE,
+			'/wizard',
+			[
+				'methods'             => [ Server::CREATABLE ],
+				'callback'            => [ $this, 'handle' ],
+				'permission_callback' => [ $this, 'check_permissions' ],
+				'args'                => [
+					'action_nonce' => [
+						'type'              => 'string',
+						'description'       => __( 'The action nonce for the request.', 'the-events-calendar' ),
+						'required'          => true,
+						'validate_callback' => [ $this, 'check_nonce' ],
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Check the nonce.
+	 *
+	 * @since 7.0.0
+	 *
+	 * @param string $nonce The nonce.
+	 *
+	 * @return bool|WP_Error True if the nonce is valid, WP_Error if not.
+	 */
+	public function check_nonce( $nonce ) {
+		$verified = wp_verify_nonce( $nonce, self::NONCE_ACTION );
+
+		if ( $verified ) {
+			return true;
+		}
+
+		return new WP_Error(
+			'tec_invalid_nonce',
+			__( 'Invalid nonce.', 'the-events-calendar' ),
+			[ 'status' => 403 ]
+		);
+	}
```

Comment:
Why nonce on REST context ? Don't we have current_user_can in this context since the script loads in a page where wp displays it to `manage_options` users ?

Reply:
APIs are by default public - more security is not a bad thing.